wcfsetup/install/files/lib/data/user/rank/ViewableUserRankList.class.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace wcf\data\user\rank;
+
+use wcf\data\TMultilingualContentList;
+use wcf\system\WCF;
+
+/**
+ * @author Olaf Braun
+ * @copyright 2001-2026 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since 6.2
+ *
+ * @extends UserRankList<ViewableUserRank>
+ */
+class ViewableUserRankList extends UserRankList
```

Comment:
I believe this whole class is pointless if you instead add all the localization data into the cache and then dynamically select the proper value. Removes the entire need to a dedicated list class and I strongly dislike those `Viewable*` classes in general, they create more problems than they solve.